Proposition of fix for #1213

Phyks opened the pull request:
Returns path relative to the catalog root path in Subsonic API.

<del>Note that for this I needed to implement the `Catalog::_get_info`
function, which was already called in the code, but was not declared
anywhere. This implementation may not be the best one…</del>

Closes issue #1213.

Afterster commented:
Your fix is good thanks for taking care of this issue.

But is `Catalog::_get_info` method really mandatory? Don't you already have `path` member variable already defined at this point without calling this method?

Phyks commented:
Indeed, I should have it defined already at this point. The call for the `Catalog::_get_info` method was there, but not implemented, so I just implemented it in a symmetrical way with respect to other classes.

I will push a commit to remove this call as it should not be needed at all. Should I still keep the implementation of the `Catalog::_get_info` method?

Phyks commented:
Ok, so I updated my PR. I removed the call to `_get_info`. Also, I noticed that there was both a `get_info` and a `_get_info` method in `Catalog` class. So I removed the `_get_info` method and updated the other catalogs which were referencing it accordingly.

Afterster commented:
Looks good for me now, thanks!
